lib/bleno.js

```diff
@@ -172,28 +209,42 @@ Bleno.prototype.startAdvertisingWithEIRData = function(advertisementData, scanDa
     scanData = null;
   }
 
-  if (this.state !== 'poweredOn') {
-    var error = new Error('Could not advertising scanning, state is ' + this.state + ' (not poweredOn)');
+  var advertise = function(state) {
 
-    if (typeof callback === 'function') {
-      callback(error);
+    if (state !== 'poweredOn') {
+      var error = new Error('Could not advertising scanning, state is ' + state + ' (not poweredOn)');
+
+      if (typeof callback === 'function') {
+        callback(error);
+      } else {
+        throw error;
+      }
     } else {
-      throw error;
-    }
-  } else {
-    if (callback) {
-      this.once('advertisingStart', callback);
+      if (callback) {
+        this.once('advertisingStart', callback);
+      }
+
+      this._bindings.startAdvertisingWithEIRData(advertisementData, scanData);
     }
+  }
 
-    this._bindings.startAdvertisingWithEIRData(advertisementData, scanData);
+  //if bindings still not init, do it now
+  if (!this.initialized) {
+    this._bindings.init();
+    this.initialized = true;
+    this.once('stateChange', advertise.bind(this));
+  }else{
+    advertise.call(this, this._state);
   }
 };
 
 Bleno.prototype.stopAdvertising = function(callback) {
   if (callback) {
     this.once('advertisingStop', callback);
   }
-  this._bindings.stopAdvertising();
+  if(this._bindings && this.initialized){
+    this._bindings.stopAdvertising();
+  }
```

Comment:
There are a few spots like this below, where an edge case exists. If the bindings are not initialized for whatever reason. The passed in callback would never be called.

Do we want to cover this case? I'm guessing something similar would happen before though ...